rasa/shared/utils/common.py

```diff
@@ -45,13 +41,10 @@ def class_from_module_path(
         raise ImportError(f"Cannot retrieve class from path {module_path}.")
 
     if not inspect.isclass(klass):
-        rasa.shared.utils.io.raise_deprecation_warning(
+        raise TypeError(
```

Comment:
I'm wondering if it's not better to raise a `RasaException` here, wouldn't this be an usage error? 🤔 
Also the docstring should be updated with the 2nd error raised other than `ImportError`.

Reply:
Good point, RasaException does not trigger the telemetry. I guess then we would want to remove the ImportError as well

Reply:
Maybe it's based to revert this change for `ImportError` based on the latest failed test information 😅 
I'm actually reviewing a PR that is filtering `ImportError` exceptions from being sent to Sentry so that will be taken care of too indirectly.

Reply:
😅 agreed